docs(web modeler): Milestones and process application changelist (#4107)
camunda/camunda-docs: docs/components/modeler/web-modeler/milestones.md

:::note

A new milestone is also automatically created when dragging and dropping a file into the diagram view, or when you select the **Replace via upload** option from the breadcrumb menu.

:::

### Process application bulk milestone creation

A [process application](/components/modeler/web-modeler/process-applications.md) is a special type of Web Modeler folder that allows you to work on a set of related files and
[deploy](/components/modeler/web-modeler/process-applications.md#deploy-and-run-a-process-application) them as a single bundle with just one click. This reduces the risk of having a broken deployment at runtime, and makes it more convenient to deploy related files.

If you create a milestone on the main process of a process application, milestones are created for all other assets in the application, to make it easier to track or revert changes.

:::note
Milestones of resources belonging to a process application are tied to the main process and cannot be modified.
:::

## Compare milestones

You can compare the change history between two milestones, either visually as a diagram or as code in an XML diff layout.

1. Open the milestone history for your diagram.

> 1. Ensure that the **Show changes** toggle in the **Changes** panel is turned on.

Reviewer: 🔧 The toggle is present in the sidebar, not specifically inside the changes panel.

Author: @nithinssabu Can I just check here: For me the toggle is in the Changes panel on the side?

![image](https://github.com/user-attachments/assets/5cd9e0f4-2de2-454c-92fd-7d700b6b796c)

Author: Changes made (sidebar)

To view BPMN diagram changes visually, select the **Visual view** tab.

![milestones diffing](img/milestones/web-modeler-milestone-visual-diffing.png)

- Differences between the milestones are highlighted visually on the diagram. For example, if an element was added, this change is highlighted in green with a plus symbol. Hover over a change to view more details.
- Only differences that affect the execution of the BPMN process are highlighted.
- The **Changes** panel shows the details of each change, including the type and identifier. Select a change to highlight it.

:::note

You can only use the **Code view** to compare changes in a DMN diagram. The **Visual view** only shows a view of the milestone.

:::

### Compare milestones in code view

To view BPMN and DMN diagram changes as code in an XML diff layout, select the **Code view** tab.

![milestones diffing](img/milestones/web-modeler-milestone-code-diffing.png)

- The XML for the previous milestone is shown on the left, with the currently selected milestone shown on the right.
- Differences between the milestones are highlighted in the XML. For example, if an element was added, this change is highlighted in green.

> - Only differences that affect the execution of the BPMN process are highlighted.

Reviewer: ❌ This is not true. All changes between the two XMLs are shown.

Author: Change made

### Compare process application milestones

> You can also use the **Code view** tab to view and compare the [versioned milestones](process-applications.md#versioning) of individual files in a process application.

Reviewer: ❌ It is wrong to say "You can use the Code view tab to view and compare milestones of individual files"

For process applications, when a user opens the milestone history for any file, the history page includes a file switcher which can be used to switch between all files in the process application to see their individual milestones.

❌ The title `Compare process application milestones` also does not make sense.

There is no concept of process application milestone. Within process applications, there is the concept of bulk creation of milestones. If a user tries to create a milestone for an individual resource, milestones are created for all resources present in the process application at that time.

❌ I also think this whole section makes more sense to be included in the process applications page under versioning to avoid confusion. There is some limitation wrt process applications milestones creation which can make it confusing. 
- Users might expect versioning of the process application (this does not exist currently)
- What exists is bulk creation of milestones so that users don't have to go to individual files and create milestones separately to save a snapshot.

Check out the [kick-off document](https://docs.google.com/document/d/1jQvjZQkU1mi5WKkb6yq0Cty4fXLlfozynh4j1tK2hto/edit#heading=h.lntg56ljm653) to get a better understanding of the limitation I'm talking about. We need to make this limitation clear through the documentation.

Author: Thanks, I'll review this whole section 👍

Reviewer: I find it really nice how you revised it ❤️

Author: @nithinssabu I've modified the versioning section following your comments - please review 👍

When you view the milestone history for any file in a process application, you can also navigate between the files to view and compare their milestone history.

![milestones restore](img/milestones/web-modeler-milestone-compare-process-application-files-highlight.png)

:::note

You cannot compare two process application versions using the **Code view**, you can only view and compare milestones for each individual process application file.

:::

## Restore a milestone

You can restore a milestone to revert to a previous snapshot of your diagram.

1. In the **Milestones** panel, hover over the milestone you want to restore.
1. Select the three vertical dots to open the actions menu.
1. Select **Restore as latest**.

![milestones restore](img/milestones/web-modeler-milestone-restore-highlight.png)

> The diagram reverts to the restored milestone. A new change milestone is created with "(restored)" appended to the name.

Reviewer: 🔧 
```suggestion
The diagram reverts to the restored milestone. A new milestone is created with "(restored)" appended to the name.
```

Author: Change made

![milestones restore](img/milestones/web-modeler-milestone-restore-complete-highlight.png)

## Copy a diagram milestone

> You can make a copy of a diagram for a specific milestone.

Reviewer: 🔧 
```suggestion
You can create a new diagram from a specific milestone.
```

Author: Edited the Copy section to reflect this change in terminology.
